src/builtin.cpp

```diff
@@ -2451,7 +2453,7 @@ static int builtin_status(parser_t &parser, io_streams_t &streams, wchar_t **arg
     /// the non-flag subcommand form. While these flags are deprecated they must be supported at
     /// least until fish 3.0 and possibly longer to avoid breaking everyones config.fish and other
     /// scripts.
-    const wchar_t *short_options = L":cbilfnhj:t";
+    const wchar_t *short_options = L":bcfhij:lnut";
```

Comment:
Simply add the new subcommand to the `status_enum_map` like you've already done and omit defining flags as an alternative. In general we're trying to switch `history`, `status`, etc. from a `command --subcommand` idiom to `command subcommand` (like `git log`).

Legacy subcommands need to continue to allow the flag form for backward compatibility but we don't want to define flags for new subcommands.